app/services/atmosphere/cloud/vm_creator.rb

```diff
@@ -27,7 +20,26 @@ def execute
         server_params[:nics] = [{ net_id: @nic }]
       end
 
-      set_security_groups!(server_params)
+      case (@tmpl.compute_site.technology)
+      when 'azure'
+        server_params[:vm_name] = SecureRandom.uuid
+        server_params[:vm_user] = Atmosphere.azure_vm_user
+        server_params[:password] = Atmosphere.azure_vm_password
```

Comment:
I am aware of this but I could not think of anything better that would not introduce a lot of work.